File: dotnet/DesignPatterns/BehavioralDesignPatterns/CommandDesignPattern/CommandDesignPattern.Example2/IncreaseBrightnessCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CommandDesignPattern.Example2;

public class IncreaseBrightnessCommand : ICommand
{
    private Light _light;
    public IncreaseBrightnessCommand(Light light)
    {
        _light = light;
    }
    public void Execute()
    {
        _light.IncreaseBrightness();
    }
    public void Undo()
    {
        _light.DecreaseBrightness();
    }
}

File: java/DesignPatterns/BehavioralDesignPatterns/CommandDesignPattern/CommandDesignPattern.Example2/src/main/java/behavioral/command/example2/IncreaseBrightnessCommand.java
package behavioral.command.example2;

public class IncreaseBrightnessCommand implements ICommand {
    private final Light light;
    public IncreaseBrightnessCommand(Light light) { this.light = light; }

    @Override
    public void execute() { light.increaseBrightness(); }

    @Override
    public void undo() { light.decreaseBrightness(); }
}
